### PR TITLE
Add downstream pipelines status auto-refresh

### PR DESCRIPTION
**Pull request by vsebe**
Add auto-refresh of the downstream jobs status summary (optional,
disabled by default). When enabled, it displays the current status of
the downstream builds as they progress with a small delay.
The auto-refresh rate is set to 5 minutes by default.
Reduce verboseness on auto-refresh: remove elapsed time info in the
get_downstream_builds() function.

[ci skip]

Signed-off-by: Violeta Sebe <vsebe@ca.ibm.com>

### CODE REVIEW

**Comment by vsebe** [comment]
jenkins compile zlinux jdk8

**Comment by vsebe** [comment]
This might require in-script approval.

File: buildenv/jenkins/common/pipeline-functions.groovy [hunk 1]
```diff
@@ -542,9 +541,6 @@ def get_downstream_builds(upstreamBuild, upstreamJobName, downstreamJobNames) {
         }
     }
 
-    elapsedTime = (System.currentTimeMillis() - startTime)/1000
-    echo "Time spent fetching downstream jobs for ${upstreamJobName}: ${elapsedTime} seconds"
-
```
[the remark this answers is not in the document]
Comment (vsebe) [reply buildenv/jenkins/common/pipeline-functions.groovy]: When auto-refresh is enabled, this would be printed every 5 mins.

**Comment by vsebe** [comment]
jenkins compile zlinux jdk8

**Comment by vsebe** [comment]
https://ci.eclipse.org/openj9/job/PullRequest-OpenJ9/2064/console 
```
Scripts not permitted to use method org.jvnet.hudson.plugins.groovypostbuild.GroovyPostbuildRecorder$BadgeManager getBuild. Administrators can decide whether to approve or reject this signature.
```

**Comment by AdamBrousseau** [comment]
https://bugs.eclipse.org/bugs/show_bug.cgi?id=553682

**Comment by vsebe** [comment]
jenkins compile zlinux jdk8

**Comment by vsebe** [comment]
jenkins compile zlinux jdk8

**Comment by vsebe** [comment]
jenkins compile zlinux jdk8

**Comment by vsebe** [comment]
jenkins compile zlinux jdk8

**Comment by vsebe** [comment]
jenkins compile zlinux jdk8

**Comment by vsebe** [comment]
jenkins compile zlinux jdk8

**Comment by vsebe** [comment]
jenkins compile zlinux jdk8

File: buildenv/jenkins/common/pipeline-functions.groovy (hunk 1 above)
[the remark this answers is not in the document]
Comment (vsebe) [reply buildenv/jenkins/common/pipeline-functions.groovy]: re-added

**Comment by vsebe** [comment]
`Scripts not permitted to use method groovy.lang.GroovyObject getMetaClass. Administrators can decide whether to approve or reject this signature.`
https://bugs.eclipse.org/bugs/show_bug.cgi?id=553778

**Comment by vsebe** [comment]
jenkins compile zlinux jdk8

File: buildenv/jenkins/jobs/pipelines/Pipeline-Build-Test-All.groovy [hunk 2]
```diff
@@ -579,3 +594,28 @@ def get_sorted_releases() {
...
+
+def draw_summary_table() {
+    def table = get_summary_table(BUILD_IDENTIFIER)
+
+    if (table) {
+        if (ENABLE_SUMMARY_AUTO_REFRESH) {
+            def actions = manager.build.actions
+            for (int i = 0; i < actions.size(); i++) {
+                 def action = actions.get(i)
+                 if (action.metaClass && action.metaClass.hasProperty(action, "text") && action.text.contains("Downstream Jobs Status")) {
```
Comment (AdamBrousseau) [inline buildenv/jenkins/jobs/pipelines/Pipeline-Build-Test-All.groovy]: Can we avoid some script approvals by reducing the amount of conditions here, or will we hit NPEs?

Comment (vsebe) [reply buildenv/jenkins/jobs/pipelines/Pipeline-Build-Test-All.groovy]: NPE

**Pull request merged**